dotCMS/src/main/java/com/dotmarketing/common/reindex/IndexResource.java

```diff
@@ -0,0 +1,381 @@
+
+        sendAdminMessage(message, MessageSeverity.INFO, 10000);
+
+        return Response.ok(new ResponseEntityView(message)).build();
+
+
+    }
+    @CloseDBIfOpened
+    @POST
+    @JSONP
+    @NoCache
+    @Path("/optimize")
+    public Response optimizeIndices(@Context final HttpServletRequest request, @Context final HttpServletResponse response) {
+        final InitDataObject init = validateUser(request, response);
+        ContentletIndexAPI api = APILocator.getContentletIndexAPI();
+        List<String> indices = api.listDotCMSIndices();
+        api.optimize(indices);
+        String message = Try.of(()->LanguageUtil.get(APILocator.getCompanyAPI().getDefaultCompany(),"message.cmsmaintenance.cache.indexoptimized")).get();
+
+
+        sendAdminMessage(message, MessageSeverity.INFO, 0);
+        return Response.ok().build();
+    }
+
+    @CloseDBIfOpened
+    @DELETE
+    @JSONP
+    @NoCache
+    @Path("/cache")
+    public Response flushIndiciesCache(@Context final HttpServletRequest request, @Context final HttpServletResponse response) {
+        final InitDataObject init = validateUser(request, response);
+        ContentletIndexAPI api = APILocator.getContentletIndexAPI();
+        List<String> indices = api.listDotCMSIndices();
+        
+        
+        Map<String, Integer> data = APILocator.getESIndexAPI().flushCaches(indices);
+        String message = Try.of(()->LanguageUtil.get(APILocator.getCompanyAPI().getDefaultCompany(),"maintenance.index.cache.flush.message")).get();
+        
+        
+        
+        
+        message=message.replace("{0}", String.valueOf(data.get("successfulShards")));
+        message=message.replace("{1}", String.valueOf(data.get("failedShards")));
+        
+        
+        sendAdminMessage(message, MessageSeverity.INFO, 5000);
+        
+        
+        
+        return Response.ok(new ResponseEntityView(data)).build();
+
+    }
+
+    private InitDataObject validateUser(final HttpServletRequest request, final HttpServletResponse response) {
+        return new WebResource.InitBuilder(request, response).requiredRoles(Role.CMS_ADMINISTRATOR_ROLE)
+                        .requiredPortlet("maintenance").init();
+
+    }
+
+    @CloseDBIfOpened
+    @GET
+    @JSONP
+    @NoCache
+    @Path("/cluster")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response getClusterStats(@Context final HttpServletRequest request, @Context final HttpServletResponse response)
+                    throws DotDataException {
+
+        validateUser(request, response);
+
+
+        final ESIndexAPI esIndexAPI = new ESIndexAPI();
+        final ClusterStats clusterStats = esIndexAPI.getClusterStats();
+
+        Builder<String, Object> builder =
+                        ImmutableMap.<String, Object>builder().put("clusterName", clusterStats.getClusterName());
+
+        for (NodeStats stats : clusterStats.getNodeStats()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'stats' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)